python/lsst/resources/http.py

```diff
@@ -1325,14 +1325,43 @@ async def get_client_session(**kwargs: Any) -> ClientSession:
                 TCP connections to the server.
             """
             if HttpResourcePath._tcp_connector is None:
-                HttpResourcePath._tcp_connector = TCPConnector(ssl=self._config.ssl_context)
+                HttpResourcePath._tcp_connector = TCPConnector(
+                    # SSL context equipped with client credentials and
+                    # configured to validate server certificates.
+                    ssl=self._config.ssl_context,
+                    # Total number of simultaneous connections this connector
+                    # keeps open with any host.
+                    #
+                    # The default is 100 but we deliberately reduced it to
+                    # avoid keeping a large number of open connexions to file
+                    # servers when thousands of quanta execute simultaneously.
+                    #
+                    # In any case, new connexions are automatically established
+                    # when needed.
+                    limit=10,
+                    # Number of simultaneous connections to a single host:port.
+                    limit_per_host=1,
+                    # Time to keep open connections alive (unit is likely
+                    # seconds, even if not documented). The default is 15.0
+                    keepalive_timeout=15.0,
+                )
 
-            return ClientSession(connector=HttpResourcePath._tcp_connector, **kwargs)
+            connect_timeout, read_timeout = self._config.timeout
+            return ClientSession(
+                connector=HttpResourcePath._tcp_connector,
+                timeout=ClientTimeout(
+                    connect=connect_timeout,
+                    sock_connect=connect_timeout,
+                    sock_read=read_timeout,
```

Comment:
This seems a bit long, potentially leaving a process stuck waiting for an actually-broken HTTP connection for 25 minutes.

But since you've been using this same read timeout for a long time with `requests`, it must not be an issue in practice.

Reply:
Yes, we have not seen issues with this in practice. We had to increase this timeout to cope with the non-understood issues related to our usage of `fsspec`. We may decrease that read timeout when we understand the root cause of the issue.